src/plugins/data_source/server/plugin.ts

```diff
@@ -3,19 +3,26 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import { first } from 'rxjs/operators';
+
 import { PluginInitializerContext, CoreSetup, CoreStart, Plugin, Logger } from 'src/core/server';
-import { dataSource, credential } from './saved_objects';
+import { dataSource, credential, CredentialSavedObjectsClientWrapper } from './saved_objects';
+import { DataSourcePluginConfigType } from '../config';
 
 import { DataSourcePluginSetup, DataSourcePluginStart } from './types';
 
+import { CryptographyClient } from './cryptography';
+
 export class DataSourcePlugin implements Plugin<DataSourcePluginSetup, DataSourcePluginStart> {
   private readonly logger: Logger;
+  private initializerContext: PluginInitializerContext<DataSourcePluginConfigType>;
 
   constructor(initializerContext: PluginInitializerContext) {
+    this.initializerContext = initializerContext;
     this.logger = initializerContext.logger.get();
   }
 
-  public setup(core: CoreSetup) {
+  public async setup(core: CoreSetup) {
     this.logger.debug('data_source: Setup');
 
     // Register credential saved object type
```

Comment:
are we going to register credential save object type here?

Reply:
> are we going to register credential save object type here?

Yes, as discussed https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2120